doc/contributing/docs/examples.rst

```diff
@@ -1,72 +1,112 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Documenting function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    Create and start a ``my_fiber`` object. The object is created and begins to
-    run immediately.
+We describe functions of Tarantool modules via Sphinx directives ``.. module::``
+and ``.. function::``:
 
-    :param function: the function to be associated with the ``my_fiber`` object
-    :param function-arguments: what will be passed to function
+..  code-block:: rst
 
-    :return: created ``my_fiber`` object
-    :rtype: userdata
+    ..  module:: fiber
 
-    **Example:**
+    ..  function:: create(function [, function-arguments])
 
-    .. code-block:: tarantoolsession
+        Create and start a fiber. The fiber is created and begins to run immediately.
 
-        tarantool> my_fiber = require('my_fiber')
-        ---
-        ...
-        tarantool> function function_name()
-                 >   my_fiber.sleep(1000)
-                 > end
-        ---
-        ...
-        tarantool> my_fiber_object = my_fiber.create(function_name)
-        ---
-        ...
+        :param function: the function to be associated with the fiber
+        :param function-arguments: what will be passed to function
 
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-               Module, class and method
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        :return: created fiber object
+        :rtype: userdata
 
-Here is an example of documenting a module (``my_box.index``), a class
-(``my_index_object``) and a function (``my_index_object.rename``).
+        **Example:**
 
-.. module:: my_box.index
+        ..  code-block:: tarantoolsession
 
-.. class:: my_index_object
+            tarantool> fiber = require('fiber')
+            ---
+            ...
+            tarantool> function function_name()
+                     >   print("I'm a fiber")
+                     > end
+            ---
+            ...
+            tarantool> fiber_object = fiber.create(function_name); print("Fiber started")
+            I'm a fiber
+            Fiber started
+            ---
+            ...
 
-    .. method:: rename(index-name)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Documenting class method and data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-        Rename an index.
+Here is an example of documenting a method and a data of a class ``index_object``:
 
-        :param index_object: an object reference
-        :param index_name: a new name for the index (type = string)
+..  code-block:: rst
 
-        :return: nil
+    ..  class:: index_object
 
-        Possible errors: index_object does not exist.
+        ..  method:: get(key)
 
-        **Example:**
+            Search for a tuple via the given index, as described :ref:`here <box_index-note>`.
```

Comment:
```rst
 :ref:`here <box_index-note>`
```
[Here](example.com) is a bad example of a link. It's much better to describe the link's target with its text:
```rst
Search for a tuple :ref:`via the given index <box_index-note>`.
```